src/extractor/suffix_table.cpp

```diff
@@ -0,0 +1,48 @@
+#include "extractor/suffix_table.hpp"
+
+#include "util/lua_util.hpp"
+#include "util/simple_logger.hpp"
+
+#include <boost/assert.hpp>
+#include <boost/ref.hpp>
+#include <boost/algorithm/string.hpp>
+
+#include <vector>
+
+namespace osrm
+{
+namespace extractor
+{
+
+void SuffixTable::fill(lua_State *lua_state)
+{
+    BOOST_ASSERT(lua_state != nullptr);
+    if (!util::luaFunctionExists(lua_state, "get_name_suffix_list"))
+        return;
+
+    std::vector<std::string> suffixes_vector;
+    try
+    {
+        // call lua profile to compute turn penalty
+        luabind::call_function<void>(lua_state, "get_name_suffix_list",
+                                     boost::ref(suffixes_vector));
+    }
+    catch (const luabind::error &er)
+    {
+        util::SimpleLogger().Write(logWARNING) << er.what();
```

Comment:
well. this is why I check for the function before. If you don't provide it, it does not fail, does it?